### PR TITLE
adjusted pom & workspace class in order to ease AEM integration + added unit tests using wiremock

### PR DESCRIPTION
**Pull request by ens36777**
## Description
adjusted pom & workspace class in order to ease AEM integration
added unit-tests using wiremock

## Related Issue
https://jira.corp.adobe.com/browse/CI-4852
https://git.corp.adobe.com/adobeio/aem-event-proxy-skyline/pull/47#discussion_r4526799

## How Has This Been Tested?
manual testing

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [X] My code follows the [code style](CODE_STYLE.md) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by francoisledroff — APPROVED** [review]

**Comment by ens36777** [comment]
@francoisledroff I've added unit-tests using wiremock, please have a look

File: events_ingress/src/test/java/com/adobe/event/publish/PublishServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+//    /** publishRawEvent */
+    stubFor(post(anyUrl())
+            .withHeader(TestUtil.CONTENT_TYPE_HEADER, equalTo(TestUtil.CONTENT_TYPE_HEADER_VALUE))
+            .withHeader("x-adobe-event-provider-id", equalTo(PROVIDER_ID))
+            .withHeader("x-adobe-event-code", equalTo(EVENT_CODE))
+            .willReturn(aResponse()
+                    .withStatus(200)
+            ));
+    getPublishService().publishRawEvent(PROVIDER_ID, EVENT_CODE, EVENT_PAYLOAD);
+    verify(postRequestedFor(anyUrl()));
```
Comment (francoisledroff) [inline events_ingress/src/test/java/com/adobe/event/publish/PublishServiceTest.java:47]: why using anyUrl and not the publish endpoint (which is the root) ... same below

File: events_journal/src/test/java/com/adobe/event/journal/JournalServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+//    /** get */
+    stubFor(get(anyUrl())
+            .withHeader(TestUtil.IMS_ORG_ID_HEADER, equalTo(TestUtil.IMS_ORG_ID))
+            .willReturn(aResponse()
+                    .withStatus(200)
+            ));
+
+    getJournalService().get(TestUtil.getWorkspace().getImsUrl());
+    verify(getRequestedFor(anyUrl()));
```
Comment (francoisledroff) [inline events_journal/src/test/java/com/adobe/event/journal/JournalServiceTest.java:43]: please enrich the sub with populated json response so you could assert the resulting `JournalEntry` object

File: events_journal/src/test/java/com/adobe/event/journal/JournalServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    //    /** getLatest */
+    stubFor(get(urlEqualTo("/?latest=true"))
+            .withHeader(TestUtil.IMS_ORG_ID_HEADER, equalTo(TestUtil.IMS_ORG_ID))
+            .willReturn(aResponse()
+                    .withStatus(200)
+            ));
+
+    getJournalService().getLatest();
+    verify(getRequestedFor(urlEqualTo("/?latest=true")));
```
Comment (francoisledroff) [inline events_journal/src/test/java/com/adobe/event/journal/JournalServiceTest.java:53]: same here, please enrich the sub with populated json response so you could assert the resulting JournalEntry object

File: events_ingress/src/test/java/com/adobe/event/publish/PublishServiceTestDrive.java [hunk 4]
```diff
@@ -14,14 +14,15 @@
...
 import com.adobe.util.JacksonUtil;
-import com.adobe.ims.util.PrivateKeyBuilder;
 import feign.RequestInterceptor;
-import java.security.PrivateKey;
-import java.util.Properties;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import java.security.PrivateKey;
+import java.util.Properties;
+
```
Comment (francoisledroff) [inline events_ingress/src/test/java/com/adobe/event/publish/PublishServiceTestDrive.java:25]: could you revert to the exact same version ? of the `*TestDrive` classes
so we avoid seeing these formatting  only changes ?

File: events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+            .willReturn(okJson("{\"_links\":{\"self\":{\"href\":\"localhost:9999/events/someConsumerOrgId/providers\"}},\"_embedded\":{\"providers\":[{\"_links\":{\"rel:eventmetadata\":{\"href\":\"localhost:9999/events/providers/207aad29-f183-4c3d-87b5-22b21b8f0ee9/eventmetadata\"},\"self\":{\"href\":\"localhost:9999/events/providers/207aad29-f183-4c3d-87b5-22b21b8f0ee9\"}},\"id\":\"207aad29-f183-4c3d-87b5-22b21b8f0ee9\",\"label\":\"Analytics Triggers\",\"description\":\"Adobe Analytics Triggers events provider (gain insight into behavior of users on your site)\",\"source\":\"urn:uuid:207aad29-f183-4c3d-87b5-22b21b8f0ee9\",\"docs_url\":\"https://localhost.com/test\",\"publisher\":\"Adobe\"}]}}")
+                    .withStatus(200)
+            ));
+    List<Provider> providers = getProviderService().getProviders();
+    verify(getRequestedFor(urlEqualTo("/events/someConsumerOrgId/providers")));
+    assertNotNull(providers);
+    assertEquals(1, providers.size());
+
+//    /** findProviderById */
+    stubFor(get(urlEqualTo("/events/providers/619363ad-2461-4c1c-bb09-848f8b4d0a30?eventmetadata=true"))
```
Comment (francoisledroff) [inline events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java:67]: use your  `PROVIDER_ID` constant here instead

File: events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+import org.junit.Test;
+import java.util.*;
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertNotNull;
+
+public class ProviderServiceTest {
+
+  private static final String PROVIDER_ID = "619363ad-2461-4c1c-bb09-848f8b4d0a30";
+  private static final String SOURCE_ID = "urn:uuid:619363ad-2461-4c1c-bb09-848f8b4d0a30";
```
Comment (francoisledroff) [inline events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java:31]: `SOURCE_ID = "urn:uuid:" + PROVIDER_ID `

File: events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+  private static final String PROVIDER_MODEL_LABEL = "aio-lib-java Wiremock - test drive provider label";
+  private static final String PROVIDER_MODEL_DESCRIPTION = "aio-lib-java Wiremock - test drive provider description";
+  private static final String PROVIDER_MODEL_DOCS_URL = "https://localhost.com/test";
+
+  private static final String METADATA_EVENTCODE = "wiremock_eventcode";
+  private static final String METADATA_DESCRIPTION = "wiremock test description";
+  private static final String METADATA_LABEL = "test label";
+
+  @Rule
+  public WireMockRule wireMockRule = new WireMockRule(9999);
```
Comment (francoisledroff) [inline events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java:42]: the wiremock port could also be a constant as it's found in the response

File: events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+
+    ProviderInputModel providerCreateModel = ProviderInputModel.builder()
+            .label(PROVIDER_MODEL_LABEL)
+            .description(PROVIDER_MODEL_DESCRIPTION)
+            .docsUrl(PROVIDER_MODEL_DOCS_URL)
+            .build();
+
+//    /** getProviders */
+    stubFor(get(urlEqualTo("/events/someConsumerOrgId/providers"))
+            .willReturn(okJson("{\"_links\":{\"self\":{\"href\":\"localhost:9999/events/someConsumerOrgId/providers\"}},\"_embedded\":{\"providers\":[{\"_links\":{\"rel:eventmetadata\":{\"href\":\"localhost:9999/events/providers/207aad29-f183-4c3d-87b5-22b21b8f0ee9/eventmetadata\"},\"self\":{\"href\":\"localhost:9999/events/providers/207aad29-f183-4c3d-87b5-22b21b8f0ee9\"}},\"id\":\"207aad29-f183-4c3d-87b5-22b21b8f0ee9\",\"label\":\"Analytics Triggers\",\"description\":\"Adobe Analytics Triggers events provider (gain insight into behavior of users on your site)\",\"source\":\"urn:uuid:207aad29-f183-4c3d-87b5-22b21b8f0ee9\",\"docs_url\":\"https://localhost.com/test\",\"publisher\":\"Adobe\"}]}}")
```
Comment (francoisledroff) [inline events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java:58]: the json payload should use the (wiremock port, provider* and event* ) constant too.
let's build a helper method may be

File: events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+public class ProviderServiceTest {
+
+  private static final String PROVIDER_ID = "619363ad-2461-4c1c-bb09-848f8b4d0a30";
+  private static final String SOURCE_ID = "urn:uuid:619363ad-2461-4c1c-bb09-848f8b4d0a30";
+
+  private static final String PROVIDER_MODEL_LABEL = "aio-lib-java Wiremock - test drive provider label";
+  private static final String PROVIDER_MODEL_DESCRIPTION = "aio-lib-java Wiremock - test drive provider description";
+  private static final String PROVIDER_MODEL_DOCS_URL = "https://localhost.com/test";
+
+  private static final String METADATA_EVENTCODE = "wiremock_eventcode";
```
Comment (francoisledroff) [inline events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java:37]: METADATA_EVENTCODE -->  EVENT_CODE
METADATA_DESCRIPTION --> EVENT_DESCRIPTION
METADATA_LABEL --> EVENT_LABEL

**Review by francoisledroff — CHANGES_REQUESTED** [review]
Please merge with the latest `main` branch, I already pushed the osgi changes 
( I would advise you use another branch from your fork as well )

look at my comments below, they all boil down to the following:
* use better url (not IMS when it's not applicable) url that makes more sense and will make the understanding of our sdk easier
* populate the wiremock json response and could assert the resulted java object 
* use test constants consistentlty (use these when populating the json payload)

File: events_ingress/src/test/java/com/adobe/event/publish/PublishServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+            .willReturn(okJson(TestUtil.AUTH_RESPONSE)
+                    .withStatus(200)
+            ));
+    RequestInterceptor authInterceptor = JWTAuthInterceptor.builder()
+            .workspace(TestUtil.getWorkspace())
+            .build();
+
+    return PublishService.builder()
+            .authInterceptor(authInterceptor)
+            .url(TestUtil.getWorkspace().getImsUrl())
```
[the remark this answers is not in the document]
Comment (ens36777) [reply events_ingress/src/test/java/com/adobe/event/publish/PublishServiceTest.java]: Publish_url added

File: events_ingress/src/test/java/com/adobe/event/publish/PublishServiceTest.java [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+    verify(postRequestedFor(anyUrl()));
+
+//    /** publishCloudEvent */
+    stubFor(post(anyUrl())
+            .withHeader(TestUtil.CONTENT_TYPE_HEADER, equalTo("application/cloudevents+json"))
+            .willReturn(aResponse()
+                    .withStatus(200)
+            ));
+    getPublishService().publishCloudEvent(PROVIDER_ID, EVENT_CODE, EVENT_PAYLOAD);
+    verify(postRequestedFor(anyUrl()));
```
[the remark this answers is not in the document]
Comment (ens36777) [reply events_ingress/src/test/java/com/adobe/event/publish/PublishServiceTest.java]: assertion for the CloudEvent object has been added.

File: events_journal/src/test/java/com/adobe/event/journal/JournalServiceTest.java [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+                    .withStatus(200)
+            ));
+    RequestInterceptor authInterceptor = JWTAuthInterceptor.builder()
+            .workspace(TestUtil.getWorkspace())
+            .build();
+
+    return JournalService.builder()
+            .workspace(TestUtil.getWorkspace())
+            .authInterceptor(authInterceptor)
+            .url(TestUtil.getWorkspace().getImsUrl())
```
[the remark this answers is not in the document]
Comment (ens36777) [reply events_journal/src/test/java/com/adobe/event/journal/JournalServiceTest.java]: JOURNAL_URL has been added

File: events_journal/src/test/java/com/adobe/event/journal/JournalServiceTest.java (hunk 3 above)
Comment (ens36777) [reply events_journal/src/test/java/com/adobe/event/journal/JournalServiceTest.java:62]: JournalEntry assertions have been added.

File: events_ingress/src/test/java/com/adobe/event/publish/PublishServiceTestDrive.java (hunk 4 above)
Comment (ens36777) [reply events_ingress/src/test/java/com/adobe/event/publish/PublishServiceTestDrive.java]: *TestDrive classes have been reverted

File: events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java (hunk 6 above)
Comment (ens36777) [reply events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java]: adjusted

File: events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java (hunk 7 above)
Comment (ens36777) [reply events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java]: port constant has been added

File: events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java (hunk 8 above)
Comment (ens36777) [reply events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java]: I've created a few methods that are returning the responses which are being used more than once and adjusted them in order to use constants. Please let me know if you have another idea for this.

File: events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java (hunk 9 above)
Comment (ens36777) [reply events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceTest.java]: adjusted

**Comment by francoisledroff** [comment]
closing this for now

**Pull request closed without merging**